File: pset1/mario/mario.c
# include <stdio.h>
# include <cs50.h>

void printBlock(string s, int n);
void pyramid(int n);
int getOneToEight(void);

int main(void)
{
    int h = getOneToEight();
    pyramid(h);
}

// geoOneToEight promts user for the int between 1 and 8, included.
int getOneToEight(void)
{
    while (true)
    {
        int height = get_int("Height: ");

        // validate input
        if (height >= 1 && height <= 8)
        {
            return height;
        }
    }
}

// pyramid prints the mario pyramid trap
void pyramid(int n)
{
    // range over rows
    for (int i = 1; i <= n; i++)
    {
        printBlock(" ", n - i); // spaces
        printBlock("#", i);     // left hashes
        printBlock(" ", 2);     // gap
        printBlock("#", i);     // right hashes
        printf("\n");
    }
}


// printBlock prints n number of #
void printBlock(string s, int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%s", s);
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    i = get_my_int()
    pyramids(i)


# get wanted number
def get_my_int():
    while True:
        i = get_int("Height: ")
        if i >= 1 and i <= 8:
            # success
            return i


# print pyramids
def pyramids(n):
    for i in range(n):
        spaces = " " * (n-1-i)
        blocks = "#" * (i+1)
        print(f"{spaces}{blocks}  {blocks}")


main()
